packages/validation/src/sarif/suppressions/plugins/FingerprintSuppressor.ts

```diff
@@ -0,0 +1,31 @@
+import {Suppression, SuppressionKind, ValidationResult} from '../../../common/sarif.js';
+import {FingerprintSuppression, Suppressor} from '../types.js';
+
+export class FingerprintSuppressor implements Suppressor<FingerprintSuppression> {
+  kind: SuppressionKind = 'external';
+
+  suppressions: FingerprintSuppression[] = [];
+
+  preload(s: FingerprintSuppression[]): Promise<void> {
+    this.suppressions = s;
+    return Promise.resolve();
+  }
+
+  suppress(problem: ValidationResult): Suppression[] | Promise<Suppression[]> {
+    const suppression = this.suppressions.find(s => {
```

Comment:
This has O(n*m) complexity with unbound lists. I'd slightly optimise this given how easy it is:

```
preload() {
  const filteredSuppressions = suppressions.filter(s => s.fingerprint && s.kind === this.kind);
  this.suppressionsByFingerprint = keyBy(filteredSuppressions, fingerprint); // as Map
}

suppress() {
  const fingerprint = problem.fingerprints?.['monokleHash/v1'];
  if (!fingerprint) return [];
  const suppression = this.suppressionByFingerprint.get(fingerprint);
  // ...
}
```

This is once O(n) and then O(m)

Reply:
Nice! Challenge accepted!